classes/Media.php

```diff
@@ -624,12 +624,9 @@ public static function ieCssSplitter($compiled_css, $cache_path, $refresh = fals
         $protocol_link = Tools::getCurrentUrlProtocolPrefix();
         //return cached css
         if (!$refresh) {
-            $cached_files = scandir($cache_path);
-            foreach ($cached_files as $file) {
-                if ($file != '.' && $file != '..') {
-                    $css_url = str_replace(_PS_ROOT_DIR_, '', $protocol_link.Tools::getMediaServer('').$cache_path.DIRECTORY_SEPARATOR.$file);
-                    $splitted_css[$css_url] = 'all';
-                }
+            foreach (array_diff(scandir($cache_path), array('..', '.')) as $file) {
```

Comment:
My dubt was about to consider or not the (...) part of the loop to be
"inside" it
.. in other words  I was not sure that this part of code is evaluated more
than one time ..

I'll do some tests to prove or negate this fact asap

  venerdì 6 maggio 2016, Xavier <notifications@github.com> ha scritto:

> In classes/Media.php
> <https://github.com/PrestaShop/PrestaShop/pull/5511#discussion_r62343437>:
>
> > @@ -624,12 +624,9 @@ public static function ieCssSplitter($compiled_css, $cache_path, $refresh = fals
> >          $protocol_link = Tools::getCurrentUrlProtocolPrefix();
> >          //return cached css
> >          if (!$refresh) {
> > -            $cached_files = scandir($cache_path);
> > -            foreach ($cached_files as $file) {
> > -                if ($file != '.' && $file != '..') {
> > -                    $css_url = str_replace(_PS_ROOT_DIR_, '', $protocol_link.Tools::getMediaServer('').$cache_path.DIRECTORY_SEPARATOR.$file);
> > -                    $splitted_css[$css_url] = 'all';
> > -                }
> > +            foreach (array_diff(scandir($cache_path), array('..', '.')) as $file) {
>
> The issue with computing the array_diff within the foreach is that you
> make the code computing the same value in every loop. By taking the
> array_diff computation out of the loop and into a variable, you gain some
> valuable microseconds, I'm sure -- moreover because you computation
> performs a scandir, which touches upon the filesystem and therefore
> waaaaaay slower than any regular array creation :)
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/PrestaShop/PrestaShop/pull/5511/files/0c5d4d63367c51316bc39319151d133daf036dca#r62343437>
>


-- 
Cordialmente,
Massimo -- *the ZiZuu staff*